src/base/charstr.rs

```diff
@@ -380,6 +380,22 @@ impl<Octs: AsRef<[u8]> + ?Sized> CharStr<Octs> {
             .compose(target)?;
         target.append_slice(self.0.as_ref())
     }
+
+    /// Appends the canonical wire format representation to an octets builder.
+    pub fn compose_canonical<Target: OctetsBuilder + ?Sized>(
```

Comment:
I think this method produces the same result as `compose` only slower?

I don’t think we need an extra method here – I think using `compose` is actually better since it makes it clear that we don’t do anything special. (This is less important for char strings that don’t anything special, but for domain names it is important to signal that they are not lower cased and for consistency, using the same strategy here seems better to me.)